pkgs/desktops/plasma-5/breeze-plymouth/default.nix

```diff
@@ -8,37 +8,40 @@
   imagemagick,
   netpbm,
   perl,
-  # these will typically need to be set via an override
-  # in a NixOS context
-  nixosBranding ? false,
-  nixosName ? "NixOS",
-  nixosVersion ? "",
+  logoName ? "nixos",
+  logoFile ? null,
+  osName ? "NixOS",
+  osVersion ? "",
   topColor ? "black",
   bottomColor ? "black"
 }:
 
-let
-  logoName = "nixos";
+let 
+  validColors = [ "black" "cardboard_grey" "charcoal_grey" "icon_blue" "paper_white" "plasma_blue" "neon_blue" "neon_green" ];
 in
+  assert lib.asserts.assertOneOf "topColor" topColor validColors;
+  assert lib.asserts.assertOneOf "bottomColor" bottomColor validColors;
+  
 mkDerivation {
   name = "breeze-plymouth";
   nativeBuildInputs = [ extra-cmake-modules ];
-  buildInputs = [ plymouth ] ++ lib.optionals nixosBranding [ imagemagick netpbm perl ];
+  buildInputs = [ plymouth ] ++ lib.optionals (logoFile != null) [ imagemagick netpbm perl ];
   patches = copyPathsToStore (lib.readPathsFromFile ./. ./series);
-  cmakeFlags = lib.optionals nixosBranding [
-    "-DDISTRO_NAME=${nixosName}"
-    "-DDISTRO_VERSION=${nixosVersion}"
+  cmakeFlags = [
+    "-DDISTRO_NAME=${osName}"
+    "-DDISTRO_VERSION=${osVersion}"
     "-DDISTRO_LOGO=${logoName}"
     "-DBACKGROUND_TOP_COLOR=${topColor}"
     "-DBACKGROUND_BOTTOM_COLOR=${bottomColor}"
   ];
+  
   postPatch = ''
       substituteInPlace cmake/FindPlymouth.cmake --subst-var out
-  '' + lib.optionalString nixosBranding ''
-      cp ${nixos-icons}/share/icons/hicolor/128x128/apps/nix-snowflake.png breeze/images/${logoName}.logo.png
+  '' + lib.optionalString (logoFile != null) ''
+      cp ${logoFile} breeze/images/${logoName}.logo.png
 
       # conversion for 16bit taken from the breeze-plymouth readme
-      convert ${nixos-icons}/share/icons/hicolor/128x128/apps/nix-snowflake.png -alpha Background -background "#000000" -fill "#000000" -flatten tmp.png
-      pngtopnm tmp.png | pnmquant 16 | pnmtopng > breeze/images/16bit/${logoName}.logo.png
+      convert ${logoFile} -alpha Background -background "#000000" -fill "#000000" -flatten tmp.png
+      pngtopnm tmp.png | pnmquant 16 | pnmtopng > ${logoName}.logo.png
```

Comment:
The output of this doesn't look like it's in the right place.

Reply:
> Very sensible, I like it. Not sure why I didn't do this to begin with...
> 
> Given that the logo file gets copied into `etc` in the main module, do we actually need to put it into the theme? I think we do, which is the reason I did it, but worth checking whether we need both.

I am not sure, I somewhat assumed it was required. Nice find for the 16bit image! I'll fix this. Should I amend this commit?